src/update_engine/payload_signer.cc

```diff
@@ -292,27 +239,44 @@ bool PayloadSigner::GetRawHashFromSignature(
   }
 
   char dummy_password[] = { ' ', 0 };  // Ensure no password is read from stdin.
-  RSA* rsa = PEM_read_RSA_PUBKEY(fpubkey, NULL, NULL, dummy_password);
+  EVP_PKEY* pkey = PEM_read_PUBKEY(fpubkey, NULL, NULL, dummy_password);
   fclose(fpubkey);
-  TEST_AND_RETURN_FALSE(rsa != NULL);
-  unsigned int keysize = RSA_size(rsa);
+  TEST_AND_RETURN_FALSE(pkey != NULL);
+  size_t keysize = EVP_PKEY_get_size(pkey);
   if (sig_data.size() > 2 * keysize) {
     LOG(ERROR) << "Signature size is too big for public key size.";
-    RSA_free(rsa);
+    EVP_PKEY_free(pkey);
     return false;
   }
 
-  // Decrypts the signature.
   vector<char> hash_data(keysize);
-  int decrypt_size = RSA_public_decrypt(
-      sig_data.size(),
-      reinterpret_cast<const unsigned char*>(sig_data.data()),
+
+  // Decrypts the signature.
+  EVP_PKEY_CTX *ctx = EVP_PKEY_CTX_new(pkey, NULL);
+  if (!ctx
+      || EVP_PKEY_verify_recover_init(ctx) <= 0
+      || EVP_PKEY_CTX_set_rsa_padding(ctx, RSA_PKCS1_PADDING) <= 0
+      || EVP_PKEY_CTX_set_signature_md(ctx, EVP_sha256()) <= 0) {
+    LOG(ERROR) << "Couldn't initialise EVP_PKEY_CTX";
+    EVP_PKEY_free(pkey);
+    return false;
+  }
+
+  size_t decrypt_size = keysize;
+
+  if (!EVP_PKEY_verify_recover(ctx,
```

Comment:
`EVP_PKEY_verify_recover` returns zero or negative value on failure, so checking for zero is not enough, is it?

```suggestion
  if (EVP_PKEY_verify_recover(ctx,
```